packages/logger/src/Logger.ts

```diff
@@ -908,6 +882,41 @@ class Logger extends Utility implements ClassThatLogs {
     }
   }
 
+  /**
+   * It sets sample rate value with the following prioprity:
+   * 1. Constructor value
+   * 2. Custom config service value
+   * 3. Environment variable value
+   * 4. Default value (zero)
+   *
+   * @private
+   * @param {number} [sampleRateValue]
+   * @returns {void}
+   */
+  private setInitialSampleRate(sampleRateValue?: number): void {
```

Comment:
As we discussed on Discord, I think we could simplify this method & merge it with `configureSampling` one:

```ts
private setInitialSampleRate(sampleRateValue?: number): void {
    this.powertoolLogData.sampleRateValue = 0;
    const constructorValue = sampleRateValue;
    const customConfigValue =
      this.getCustomConfigService()?.getSampleRateValue();
    const envVarsValue = this.getEnvVarsService().getSampleRateValue();
    for (const value of [constructorValue, customConfigValue, envVarsValue]) {
      if (this.isValidSampleRate(value)) {
        this.powertoolLogData.sampleRateValue = value;

        if (value && randomInt(0, 100) / 100 <= value) {
          this.setLogLevel('DEBUG');
          this.debug('Setting log level to DEBUG due to sampling rate');
        }

        return;
      }
    }
  }
```

Note that I have also swapped the order of the `this.setLogLevel('DEBUG');` and the actual `this.debug()` call. This is because in the previous order the log was never emitted if the log level was not already `DEBUG`.

In order for this to work, we'll also have to change the order of the function calls in the `setOptions` method to this:

```ts
this.setConsole();
this.setCustomConfigService(customConfigService);
this.setInitialLogLevel(logLevel);
this.setLogFormatter(logFormatter);
this.setPowertoolLogData(serviceName, environment);
this.setInitialSampleRate(sampleRateValue);
// .. others
```

This is because in order to set the level and log, we need both the log level, the log formatter, and the Powertools log data to be set.